android/cli/commands/_build.js

```diff
@@ -3409,20 +3417,16 @@ AndroidBuilder.prototype.generateTheme = async function generateTheme() {
 	const xmlFilePath = path.join(valuesDirPath, 'ti_styles.xml');
 	this.logger.info(__('Generating theme file: %s', xmlFilePath.cyan));
 
-	// Set up all "Base.Theme.Titanium.Basic" inherited themed activities to be fullscreen if enabled in "tiapp.xml".
-	let basicParentThemeName = 'Theme.AppCompat';
-	if (this.tiapp.fullscreen || this.tiapp['statusbar-hidden']) {
-		basicParentThemeName += '.Fullscreen';
-	}
-
-	// Set up all "Base.Theme.Titanium.Customizable" inherited themes to use one of the following:
-	// - The custom theme applied to the <application/> element in "AndroidManifest.xml".
-	// - The above fullsceen enabled basic theme if custom theme was not applied.
-	let customizableParentThemeName = basicParentThemeName;
+	// Set up "Base.Theme.Titanium.Customizable" inherited themes to use <application/> defined theme, if provided.
+	// Note: Do not assign it if set to a Titanium theme, which would cause a circular reference.
+	let customizableParentThemeName = this.defaultAppThemeName;
 	if (this.customAndroidManifest) {
-		let appTheme = this.customAndroidManifest.getAppAttribute('android:theme');
-		if (appTheme && appTheme.startsWith('@style/') && (appTheme !== '@style/Theme.Titanium.Translucent')) {
-			customizableParentThemeName = appTheme.replace('@style/', '');
+		const appTheme = this.customAndroidManifest.getAppAttribute('android:theme');
+		if (appTheme) {
+			const appThemeWithoutPrefix = appTheme.replace('@style/', '');
+			if (!appThemeWithoutPrefix.startsWith('Theme.Titanium') && !appThemeWithoutPrefix.startsWith('Base.Theme.Titanium')) {
+				customizableParentThemeName = appTheme;
```

Comment:
The old code dropped the leading '@style/' from the value, and does not appear to do so anymore here.

Should this line be:
```suggestion
				customizableParentThemeName = appThemeWithoutPrefix;
```

Reply:
The leading `@style/` is optional since it's used within a `<style/>` element. We don't have to remove it. I thought it would be better to copy it over as-is. In hindsight, it might have been better to get rid of the `replace()` call and just do a `startsWith('@style/Theme.Titanium')` instead since the leading `@style/` is not optional in the "AndroidManifest.xml".

We also never strip off the prefix if it leads with `@android:style/` which is for Android OS defined styles. Not even before this PR. An exception will occur when you launch the UI if you do this because the `AppCompatActivity` class requires an `AppCompat` derived theme and the exception will state exactly that.